sdk/ml/azure-ai-ml/tests/dataset/unittests/test_data_utils.py

```diff
@@ -68,13 +69,12 @@ def test_read_mltable_metadata_contents(
             assert contents["paths"] == [OrderedDict([("file", "./tmp_file.csv")])]
 
         # remote https inaccessible
-        with pytest.raises(Exception) as ex:
+        with pytest.raises(ServiceRequestError) as ex:
```

Comment:
ServiceRequestError is a sub class of Exception.

Why the error was not caught by `with pytest.raises(Exception) as ex:`?

Reply:
It _was_ caught by that exception. I'm assuming that the original test author was trying to check that the request fails by catching the exception raised when the request fails and then checking the message returned by the exception.

Checking the exception message has revealed itself to be somewhat fragile, given how it changed on us here. `azure.core.exception.ServiceRequestError` seems to get raised when a request fails before it was sent, which is exactly what the test was checking for either way. So changing the caught exception makes this more explicit, and fixes the issue we were seeing.